tests/cli/test_suite.py

```diff
@@ -3081,3 +3087,719 @@ def test_suite_new_profile_runs_notebook_opens_jupyter(
+            "no_msg",
+            id="warning: --non-interactive --datasource-name",
+        ),
+        # Yes error expected
+        # both interactive flags, datasource_name=None, with/without batch_request
+        pytest.param(
+            True,
+            True,
+            None,
+            None,
+            True,
+            None,
+            None,
+            "error_both_interactive_flags",
+            "no_msg",
+            id="error: --interactive --non-interactive",
+        ),
+        pytest.param(
+            True,
+            True,
+            None,
+            "batch_request.json",
+            True,
+            None,
+            None,
+            "error_both_interactive_flags",
+            "no_msg",
+            id="error: --interactive --non-interactive --batch-request",
+        ),
+        # both interactive flags, datasource_name=something, with/without batch_request
+        pytest.param(
+            True,
+            True,
+            "some_datasource_name",
+            None,
+            True,
+            None,
+            None,
+            "error_both_interactive_flags",
+            "no_msg",
+            id="error: --interactive --non-interactive --datasource-name",
+        ),
+        pytest.param(
+            True,
+            True,
+            "some_datasource_name",
+            "batch_request.json",
+            True,
+            None,
+            None,
+            "error_both_datasource_name_and_batch_request_flags",
+            "no_msg",
+            id="error: --interactive --non-interactive --datasource-name --batch-request",
+        ),
+        # both --datasource-name and --batch-request
+        pytest.param(
+            False,
+            False,
+            "some_datasource_name",
+            "batch_request.json",
+            True,
+            None,
+            True,
+            "error_both_datasource_name_and_batch_request_flags",
+            "no_msg",
+            id="error: --datasource-name --batch-request",
+        ),
+        pytest.param(
+            True,
+            False,
+            "some_datasource_name",
+            "batch_request.json",
+            True,
+            None,
+            True,
+            "error_both_datasource_name_and_batch_request_flags",
+            "no_msg",
+            id="--interactive --datasource-name --batch-request",
+        ),
+        pytest.param(
+            False,
+            True,
+            "some_datasource_name",
+            "batch_request.json",
+            True,
+            None,
+            True,
+            "error_both_datasource_name_and_batch_request_flags",
+            "no_msg",
+            id="--non-interactive --datasource-name --batch-request",
+        ),
+    ],
+)
+@mock.patch("click.prompt")
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+def test__process_suite_edit_flags_and_prompt(
```

Comment:
@anthonyburdi Not sure if possible, but if `process_suite_new/edit_flags_and_prompt` can be one method, this test would get simpler (up to you).